src/ccpp_constituent_prop_mod.F90

```diff
@@ -121,12 +205,70 @@ subroutine copyConstituent(outConst, inConst)
       outConst%var_long_name = inConst%var_long_name
       outConst%vert_dim = inConst%vert_dim
       outConst%const_ind = inConst%const_ind
-      outConst%field_ind = inConst%field_ind
       outConst%advected = inConst%advected
+      outConst%const_type = inConst%const_type
+      outConst%const_water = inConst%const_water
+      outConst%min_val = inConst%min_val
+      outConst%const_default_value = inConst%const_default_value
    end subroutine copyConstituent
 
    !#######################################################################
 
+   character(len=10) function to_str(val)
+      ! return default integer as a left justified string
+
+      ! Dummy argument
+      integer, intent(in) :: val
+
+      write(to_str,'(i0)') val
+
+   end function to_str
+
+   !#######################################################################
+
+   subroutine initialize_errvars(errcode, errmsg)
+      ! Initialize error variables, if present
+
+      ! Dummy arguments
+      integer,          optional, intent(out) :: errcode
+      character(len=*), optional, intent(out) :: errmsg
+
+      if (present(errcode)) then
+         errcode = 0
+      end if
+      if (present(errmsg)) then
+         errmsg = ''
+      end if
+   end subroutine initialize_errvars
+
+   !#######################################################################
+
+   subroutine append_errvars(errcode_val, errmsg_val, errcode, errmsg)
+      ! Append to error variables, if present
+
+      ! Dummy arguments
+      integer,           intent(in)    :: errcode_val
+      character(len=*),  intent(in)    :: errmsg_val
+      integer,          optional, intent(inout) :: errcode
+      character(len=*), optional, intent(inout) :: errmsg
```

Comment:
My idea for cleaning up calls to `append_errvars` (see comment below about the calling pattern).
```suggestion
   subroutine append_errvars(errcode_val, errmsg_val, subname, errcode, errmsg, caller)
      ! Append to error variables, if present

      ! Dummy arguments
      integer,           intent(in)    :: errcode_val
      character(len=*),  intent(in)    :: errmsg_val
      character(len=*),  intent(in)    :: subname
      integer,          optional, intent(inout) :: errcode
      character(len=*), optional, intent(inout) :: errmsg
      character(len=*),  optional, intent(in)    :: caller
```